### PR TITLE
stats: introduce `Event::Dispatcher::name()` to unify per-handler/worker statistics, such as `server.worker_2.watchdog_miss`

### PR DESCRIPTION
**Pull request by yskopets**
Description: introduce `Event::Dispatcher::name()` to unify per-handler/worker statistics, such as `server.worker_2.watchdog_miss`
Risk Level: Medium
Testing: integration test, manual
Docs Changes: N/A
Release Notes: N/A
Fixes: #10544

### CODE REVIEW

File: source/common/event/dispatcher_impl.cc [hunk 1]
```diff
@@ -56,10 +58,12 @@ DispatcherImpl::~DispatcherImpl() {
 #endif
 }
 
-void DispatcherImpl::initializeStats(Stats::Scope& scope, const std::string& prefix) {
+void DispatcherImpl::initializeStats(Stats::Scope& scope,
+                                     const absl::optional<std::string>& prefix) {
+  std::string effective_prefix = prefix.has_value() ? *prefix : absl::StrCat(name_, ".");
```
Comment (mattklein123) [inline source/common/event/dispatcher_impl.cc:63]: nit: const

File: source/common/event/dispatcher_impl.h [hunk 2]
```diff
@@ -99,6 +101,7 @@ class DispatcherImpl : Logger::Loggable<Logger::Id::main>,
     return run_tid_.isEmpty() || run_tid_ == api_.threadFactory().currentThreadId();
   }
 
+  std::string name_;
```
Comment (mattklein123) [inline source/common/event/dispatcher_impl.h:104]: nit: const

File: source/common/event/dispatcher_impl.cc (hunk 1 above)
Comment (yskopets) [reply source/common/event/dispatcher_impl.cc]: fixed

File: source/common/event/dispatcher_impl.h (hunk 2 above)
Comment (yskopets) [reply source/common/event/dispatcher_impl.h]: fixed

**Pull request merged**